src/actions/acquire_single_freq_fft.py

```diff
@@ -212,47 +215,85 @@ def acquire_data(self):
 
         return data
 
-    def build_sigmf_md(self, task_id, data):
+    def build_sigmf_md(self, task_id, data, schedule_entry, start_time, end_time):
         logger.debug("Building SigMF metadata file")
 
         # Use the radio's actual reported sample rate instead of requested rate
         sample_rate = self.sdr.radio.sample_rate
+        frequency = self.sdr.radio.frequency
 
         sigmf_md = SigMFFile()
-        sigmf_md.set_global_info(GLOBAL_INFO)
+        sigmf_md.set_global_info(
+            GLOBAL_INFO.copy()
+        )  # prevent GLOBAL_INFO from being modified by sigmf
         sigmf_md.set_global_field("core:sample_rate", sample_rate)
 
-        sensor_def = capabilities["sensor_definition"]
-        sensor_def["id"] = settings.FQDN
-        sigmf_md.set_global_field("ntia-sensor:sensor", sensor_def)
+        measurement_object = {
+            "time_start": start_time,
+            "time_stop": end_time,
+            "domain": "Frequency",
+            "measurement_type": "single-frequency",
+            "frequency_tuned_low": frequency,
+            "frequency_tuned_high": frequency,
+        }
+        sigmf_md.set_global_field("ntia-core:measurement", measurement_object)
+
+        sensor = capabilities["sensor"]
+        sensor["id"] = settings.FQDN
+        get_sensor_location_sigmf(sensor)
+        sigmf_md.set_global_field("ntia-sensor:sensor", sensor)
+
+        from status.views import get_last_calibration_time
+
+        sigmf_md.set_global_field(
+            "ntia-sensor:calibration_datetime", get_last_calibration_time()
+        )
+
+        sigmf_md.set_global_field("ntia-scos:task", task_id)
 
         action_def = {
             "name": self.name,
             "description": self.description,
-            "type": ["FrequencyDomain"],
+            "summary": self.description.splitlines()[0],
         }
 
         sigmf_md.set_global_field("ntia-scos:action", action_def)
-        sigmf_md.set_global_field("ntia-scos:task_id", task_id)
+
+        from schedule.serializers import ScheduleEntrySerializer
+
+        serializer = ScheduleEntrySerializer(
+            schedule_entry, context={"request": schedule_entry.request}
+        )
+        schedule_entry_json = serializer.to_sigmf_json()
+        schedule_entry_json["id"] = schedule_entry.name
+        sigmf_md.set_global_field("ntia-scos:schedule", schedule_entry_json)
+
+        sigmf_md.set_global_field(
+            "ntia-location:coordinate_system", get_coordinate_system_sigmf()
+        )
 
         capture_md = {
-            "core:frequency": self.sdr.radio.frequency,
+            "core:frequency": frequency,
             "core:datetime": utils.get_datetime_str_now(),
```

Comment:
Should this time get pushed down to the `radio_interface` as well?

Reply:
Moved to `usrp_iface`